samples/excel/90-scenarios/currency-converter.yaml

```diff
@@ -0,0 +1,178 @@
+order: 5
+id: excel-scenarios-currency-converter
+name: Currency Converter
+description: 'Reads rows and update base currency column, based on value date.'
+author: cakriwut
+host: EXCEL
+api_set:
+    ExcelApi: '1.4'
+script:
+    content: |
+      /*
+       * Copyright (c) Riwut Libinuko. All rights reserved. Licensed under the MIT license.
+      */
+
+      declare let moment: any;
+      let tblSetup: Record<string, number> = {};
+
+      $("#convert").click(() => tryCatch(convert));
+      $("#setup").click(() => tryCatch(setupSample));      
+
+      /** Main converting function **/
+      async function convert() {
+        await Excel.run(async (context) => {
+          let range = context.workbook.getSelectedRange(); 
+          // Check if any cell inside table is selected.
+          let table = range.getTables();
+          table.load("items")    
+          await context.sync();
+          if(table.items.length == 0){
+            console.error("Error: Please select any cell inside table.");
+            return;
+          }
+          
+          let selectedTable = table.getFirst();
+          selectedTable.columns.load("items");
+          await context.sync();
+          if(!IsTableValid(selectedTable.columns.items)) {
+            console.error('Error: Some of default table header is missing. Required headers: \r\n"CCY","Price (CCY)","Trx Date","Base","Price in Base"')
+            return;
+          }
+          selectedTable.rows.load("items,values");
+          await context.sync();
+          selectedTable.rows.items.forEach(async (row,idx,arr) => {
+            let priceBaseRange = row.getRange().getCell(0, tblSetup['Price in Base']);
+            priceBaseRange.load("address,text");
+            await context.sync();
+              let priceInBase = row.values[0][tblSetup['Price in Base']];
+              let priceInCcy = row.values[0][tblSetup['Price (CCY)']];
+              let trxDate = row.values[0][tblSetup['Trx Date']]
+              let ccy = row.values[0][tblSetup['CCY']];
+              let baseCcy = row.values[0][tblSetup['Base']]
+              if (priceInBase === "") {
+                let dateMoment = moment.fromOADate(trxDate);
+                let period = dateMoment.format("YYYY-MM-DD");
+                let queryUrl = convertValue(ccy, baseCcy, period, period);
+                let response = await fetch(queryUrl);
+                if (response.ok) {
+                  let result = await response.json();
+                  let converted = priceInCcy * result.rates[baseCcy];
+                  priceBaseRange.values = [[converted]];
+                  context.sync();
```

Comment:
You need to add an `await` to this `context.sync()`. Otherwise there's a race condition (try your sample with 3+ rows).

Reply:
Updated.